### PR TITLE
Refactor `directed_hausdorff_distance` and `haversine_distance` into `GeoSeries` Interface

### LINKED ISSUE #917: [FEA]: Haversine distance should probably return a `Series`, not a `NumericalColumn`.
### Is this a new feature, an improvement, or a change to existing functionality?

Change

### How would you describe the priority of this feature request

Low (would be nice)

### Please provide a clear description of problem you would like to solve.

`cuspatial.haversine_distance` returns a `NumericalColumn`. It should probably return a `Series` so it can be used more intuitively in workflows, as `column` behavior in `cudf` is mostly internal and private.

### Describe any alternatives you have considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #920: Convert `Directed hausdorff distance` to Accept GeoSeries as Input

### LINKED ISSUE #921: Convert `Haversine distance` to Accept GeoSeries as Input

### PR DESCRIPTION
**Pull request by isVoid**
## Description

Closes #921, closes #920

Also Fixes #917 

This PR refactors all functions in `distance.py` to accept geoseries as input. It also adds two interfaces to GeoSeries: `from_points_xy` and `from_multpoints_xy`. Since geospatial data source can vary, these APIs ensures flexibility when constructing a geoseries.

This PR also removes some stale demo files that uses these APIs.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cuspatial/cuspatial/core/geoseries.py [hunk 1]
```diff
@@ -608,6 +609,53 @@ def _align_to_index(
...
+        Construct a GeoSeries of MULTIPOINTs from an array of interleaved xy
+        coordinates.
+
+        Parameters
+        ----------
+        points_xy: array-like
+            Coordinates of the points, interpreted as interlaved x-y coords.
+        geometry_offset: array-like
+            Offsets indicating the starting index of the multipoint. See
+            example for detail.
```
Comment (harrism) [inline python/cuspatial/cuspatial/core/geoseries.py:642]: What example?

**Review by harrism — APPROVED** [review]
Nice improvement. One question.

**Review by thomcom — APPROVED** [review]

File: python/cuspatial/cuspatial/core/geoseries.py (hunk 1 above)
Comment (isVoid) [reply python/cuspatial/cuspatial/core/geoseries.py]: Oops, now added.

File: python/cuspatial/cuspatial/core/geoseries.py [hunk 2]
```diff
@@ -608,6 +609,60 @@ def _align_to_index(
...
+    @classmethod
+    def from_multipoints_xy(cls, multipoints_xy, geometry_offset):
+        """
+        Construct a GeoSeries of MULTIPOINTs from an array of interleaved xy
+        coordinates.
+
+        Parameters
+        ----------
+        points_xy: array-like
+            Coordinates of the points, interpreted as interlaved x-y coords.
```
Comment (thomcom) [inline python/cuspatial/cuspatial/core/geoseries.py:639]:
```suggestion
            Coordinates of the points, interpreted as interleaved x-y coords.
```

File: python/cuspatial/cuspatial/core/_column/geocolumn.py [hunk 3]
```diff
@@ -242,6 +237,57 @@ def _from_points_xy(cls, points_xy: ColumnBase):
             meta,
         )
 
+    @classmethod
+    def _from_multipoints_xy(
+        cls, multipoints_xy: ColumnBase, geometry_offsets: ColumnBase
+    ):
+        """
+        Create a GeoColumn of only single points from a cudf Series with
```
Comment (thomcom) [inline python/cuspatial/cuspatial/core/_column/geocolumn.py:245]:
```suggestion
        Create a GeoColumn of multipoints from a cudf Series with
```

**Pull request merged**